Util/AnnotationLimitProcessor.php

```diff
@@ -9,25 +9,25 @@
 class AnnotationLimitProcessor implements LimitProcessorInterface
```

Comment:
This class is a service now, it is instantiated only once by the symfony DI. If the annotations package is not available, disabling it will not require changes in the code, only changing the `noxlogic_rate_limit.processor` alias

Reply:
Main reason to move it into the service - to create object only once via DI? 

Disabling class in DI good approach. Maybe I wrong, but it doesn't bring big value, cause if annotation package is not available we need to disable `rate_limit_annotation_listener`. And without events usage, annotation class will not be used as well. 

Before that changes, `Util/AnnotationLimitProcessor.php` were tightly coupled with symfony framework, cause its use symfony related attributes. 

So in general, I agree, that it can be service class, cause in any case it right now symfony specific.

Reply:
Having stateless services is always a good idea, in the previous implementation `AnnotationLimitProcessor` had to be instantiated in the request process, while now is instantiated via DI (with all the possible optimizations that symfony will be able to do as the upcoming preloading in 7.4).

Did not know that the aim of this bundle is to be framework agnostic.Wiring it into laravel might be possible, but not sure is the main goal.